modules/aws_asg/main.tf

```diff
@@ -332,9 +332,16 @@ resource "aws_autoscaling_policy" "workers_add_capacity" {
   estimated_instance_warmup = "${var.worker_asg_scale_out_cooldown}"
   metric_aggregation_type   = "Maximum"
 
+  # Headroom is just below THRESHOLD, scale out normally
   step_adjustment {
     scaling_adjustment          = "${var.worker_asg_scale_out_qty}"
-    metric_interval_lower_bound = 1.0
+    metric_interval_lower_bound = "${floor(var.worker_asg_scale_out_threshold/-2.0)}"
+  }
+
+  # Headroom is less than half of THRESHOLD; scale out twice as much
+  step_adjustment {
+    scaling_adjustment          = "${var.worker_asg_scale_out_qty * 2}"
+    metric_interval_upper_bound = "${floor(var.worker_asg_scale_out_threshold/-2.0)}"
```

Comment:
Do the negative interval bounds translate as drops in headroom?  What do the `metric_interval_lower_bound` and `metric_interval_upper_bound` usually look like?

Reply:
The bounds are relative to the threshold. So if the threshold is 10, and the lower bound of one action is -4, then that action will execute when the level is above 6.
If there's an action with an *upper* bound of -4, then it will execute when the level drops _below_ 6.

Reply:
For the threshold `v1.travis.rabbitmq.consumers.staging.builds.ec2.headroom < 8 for 2 minutes`

This is what it translates to: 

![step-scaling](https://user-images.githubusercontent.com/1585815/30442527-01b5bca0-997d-11e7-8535-b008954284c0.png)

If the bounds were positive, it would instead look like this:

![step-scaling](https://user-images.githubusercontent.com/1585815/30442410-9ee70c78-997c-11e7-9ea2-72160fa94fd9.png)

Reply:
I think I understand better ⁉️ 😖